src/arena.c

```diff
@@ -2054,9 +2054,11 @@ arena_new(tsdn_t *tsdn, unsigned ind, extent_hooks_t *extent_hooks) {
 		/* background_thread_create() handles reentrancy internally. */
 		if (have_background_thread) {
 			bool err;
+			malloc_mutex_unlock(tsdn, &arenas_lock);
```

Comment:
This doesn't look safe to me.  If we really need to avoid owning this lock during background thread creation, we need to do some refactoring that still allows ```arenas_lock``` to protect ```arenas``` writes.